scipy/linalg/_matfuncs.py

```diff
@@ -291,7 +291,8 @@ def expm(A):
     n = a.shape[-1]
     # Empty array
     if min(*a.shape) == 0:
-        return np.empty_like(a)
+        dtype = expm(np.eye(1, dtype=a.dtype)).dtype
+        return np.empty_like(a, dtype=dtype)
```

Comment:
Based on https://github.com/scipy/scipy/pull/20295/files#r1545784590, this looks like it's the right approach. However, the smaller the array size, the more likely it is for it to have a special case. (Indeed, there is a special 1x1 case below.) Practically, the result may be the same, but conceptually, I don't think we want a special case to match the behavior of another special case; rather, we'd want the special cases to match the behavior of the general case. So I'll change this to `np.eye(2...)` before merging, and maybe I'll add a test for the 1x1 case while I'm at it.